node/localnode.go

```diff
@@ -196,15 +195,6 @@ func (localNode *LocalNode) shouldConnectToNode(n *nnetpb.Node) error {
 		}
 	}
 
-	addr, err := url.Parse(n.GetAddr())
-	if err != nil {
-		return err
-	}
-
-	if n.GetId() != nil && !bytes.Equal(address.GenChordID(addr.Host), n.GetId()) {
```

Comment:
We should add verification logic here:
1. try to get remote node id (using nodeData.PublicKey)
2. if get id success, compare it with n.GetId(), return err if not match
3. if get id failed, check if local node is persist finished, return err if local node is persist finished

Reply:
if getid failed and local node is not persist finished, how to deal with it?

Reply:
then we should ignore it and just return nil